{{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/multisalesforce/provider.py

```diff
@@ -1,28 +1,35 @@
 from allauth.socialaccount.providers.salesforce.provider import SalesforceProvider
 
 
-class PromptLoginMixin:
+class ProviderMixin:
     def get_auth_params(self, request, action):
         ret = super().get_auth_params(request, action)
         # This will ensure that even if you're logged in to Salesforce,
         # you'll be prompted to choose an identity to auth as:
         ret["prompt"] = "login"
         return ret
 
+    def extract_uid(self, data):
+        # The SalesforceProvider in allauth assumes that user_id is unique,
+        # but it can be the same between multiple sandboxes that were
+        # copied from the same production org. So we need to add the org id
+        # too to disambiguate.
+        return "{}/{}".format(data["organization_id"], data["user_id"])
 
-class SalesforceProductionProvider(PromptLoginMixin, SalesforceProvider):
+
+class SalesforceProductionProvider(ProviderMixin, SalesforceProvider):
     id = "salesforce-production"
     name = "Salesforce Production"
     package = "{{cookiecutter.project_slug}}.multisalesforce"
 
 
-class SalesforceTestProvider(PromptLoginMixin, SalesforceProvider):
+class SalesforceTestProvider(ProviderMixin, SalesforceProvider):
     id = "salesforce-test"
     name = "Salesforce Test"
     package = "{{cookiecutter.project_slug}}.multisalesforce"
 
 
-class SalesforceCustomProvider(PromptLoginMixin, SalesforceProvider):
+class SalesforceCustomProvider(ProviderMixin, SalesforceProvider):
     id = "salesforce-custom"
     name = "Salesforce Custom"
     package = "{{cookiecutter.project_slug}}.multisalesforce"
```

Comment:
The multisalesforce allauth stuff is a good candidate for moving to sfdo-template-helpers (or submit it to allauth). However I'd like to think that through first, because the current approach has some issues (i.e. it's possible to log in as the same user via either the production provider or the custom provider, which ends up creating 2 users in Django)